plugins/modules/aci_fabric_pod_selector.py

```diff
@@ -0,0 +1,298 @@
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(
+        description=dict(type="str", aliases=["descr"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+        pod_profile=dict(type="str"),
+        name=dict(type="str", aliases=["selector", "pod_selector"]),
+        selector_type=dict(type="str", choices=["ALL", "range"]),
+        policy_group=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["pod_profile", "name", "selector_type"]],
+            ["state", "present", ["pod_profile", "name", "selector_type"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name_alias = module.params.get("name_alias")
+    pod_profile = module.params.get("pod_profile")
+    name = module.params.get("name")
+    selector_type = module.params.get("selector_type")
+    policy_group = module.params.get("policy_group")
+    description = module.params.get("description")
+    state = module.params.get("state")
+
+    child_classes = ["fabricRsPodPGrp"]
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fabricPodP",
+            aci_rn="fabric/podprof-{0}".format(pod_profile),
+            module_object=pod_profile,
+            target_filter={"name": pod_profile},
+        ),
+        subclass_1=dict(
+            aci_class="fabricPodS",
+            aci_rn="pods-{0}-typ-{1}".format(name, selector_type),
+            module_object=name,
+            target_filter={"name": name},
```

Comment:
is name only target filter required?

Reply:
I've updated the filter to include selector_type